### PR TITLE
[SPARKR-151] Add name, setName in RDD.R

### PR DESCRIPTION
**Pull request by lythesia**
SPARKR-151: Support ability to set, retrieve RDD name.

### CODE REVIEW

File: pkg/R/RDD.R [hunk 1]
```diff
@@ -1844,3 +1844,53 @@ setMethod("cogroup",
             cogroup.rdd <- mapValues(groupByKey(union.rdd, numPartitions), 
                                      group.func)
           })
+
+# TODO:
+# bind to env?
```
Comment (shivaram) [inline pkg/R/RDD.R]: Is this TODO referring to saving the name in the RDD's env ?

File: pkg/R/RDD.R [hunk 2]
```diff
@@ -1844,3 +1844,53 @@ setMethod("cogroup",
...
+#' name(rdd) # NULL (if not set before)
+#'}
+setGeneric("name", function(rdd) { standardGeneric("name") })
+
+#' @rdname name
+#' @aliases name,RDD
+setMethod("name",
+          signature(rdd = "RDD"),
+          function(rdd) {
+            .jcall(getJRDD(rdd), "S", "name")
```
Comment (shivaram) [inline pkg/R/RDD.R]: I can see that "S" is a shortcut for "Ljava/lang/String;", but I'd actually prefer having the full type here to be explicit.

File: pkg/R/RDD.R [hunk 3]
```diff
@@ -1844,3 +1844,53 @@ setMethod("cogroup",
...
+          }
+)
+
+#' Set an RDD's name.
+#'
+#' @param rdd The RDD whose name is returned
```
Comment (shivaram) [inline pkg/R/RDD.R]: This param should be `The RDD whose name is to be set`

File: pkg/R/RDD.R [hunk 4]
```diff
@@ -1844,3 +1844,53 @@ setMethod("cogroup",
...
+setGeneric("setName", function(rdd, name) { standardGeneric("setName") })
+
+#' @rdname setName
+#' @aliases setName,RDD
+setMethod("setName",
+          signature(rdd = "RDD", name = "character"),
+          function(rdd, name) {
+            .jcall(getJRDD(rdd), "Lorg/apache/spark/api/java/JavaRDD;", "setName", name)
+            rdd
+          }
```
Comment (shivaram) [inline pkg/R/RDD.R]: I think the style we use here is to put the `)` on the same line as the `}`

**Comment by shivaram** [comment]
Thanks @lythesia for the change. It mostly looks good and I left some minor inline comments.

Also could you modify the title of the PR to be like `[SPARKR-151] Add name, setName in RDD.R`

File: pkg/R/RDD.R (hunk 1 above)
Comment (lythesia) [inline pkg/R/RDD.R]: yes, like the `rdd@env$isCached`, then you may access the name without `.jcall`

File: pkg/R/RDD.R (hunk 2 above)
Comment (lythesia) [inline pkg/R/RDD.R]: Thx, I'll fix that.

File: pkg/R/RDD.R (hunk 1 above)
Comment (shivaram) [inline pkg/R/RDD.R]: We could do that, but I am not sure `name` will be that frequently used. I am fine with leaving it as a TODO -- You could word it as something like `TODO: Consider caching the name in the RDD's environment`

**Comment by shivaram** [comment]
Thanks for the update. One minor comment: The roxygen files need to be regenerated after updating the help text. Let me know if you are having trouble with it and I can regenerate it while merging

**Comment by lythesia** [comment]
Sorry just missed that, manual fixed.

**Comment by shivaram** [comment]
Thanks @lythesia. Merging this

**Pull request merged**